python_modules/dagster/dagster/_core/definitions/asset_selection.py

```diff
@@ -239,13 +240,18 @@ def groups(*group_strs, include_sources: bool = False) -> "GroupsAssetSelection"
             include_sources (bool): If True, then include external assets matching the group in the
                 selection.
         """
-        check.tuple_param(group_strs, "group_strs", of_type=str)
-        return GroupsAssetSelection(selected_groups=group_strs, include_sources=include_sources)
+        check.tuple_param(group_strs, "group_strs", of_type=Optional[str])
+        return GroupsAssetSelection(
+            selected_groups=list(filter(lambda x: x is not None, group_strs)),
+            include_sources=include_sources,
+        )
 
     @public
     @staticmethod
     @beta_param(param="include_sources")
-    def tag(key: str, value: str, include_sources: bool = False) -> "AssetSelection":
+    def tag(
+        key: Optional[str], value: Optional[str], include_sources: bool = False
```

Comment:
I think the key should always be a string -- it's just the value that's optional